queue/src/main/java/org/killbill/queue/DBBackedQueue.java

```diff
@@ -763,47 +764,33 @@ public void reapEntries(final Date reapingDate) {
         sqlDao.inTransaction(new Transaction<Void, QueueSqlDao<T>>() {
             @Override
             public Void inTransaction(final QueueSqlDao<T> transactional, final TransactionStatus status) throws Exception {
-                final Date now = clock.getUTCNow().toDate();
-                final List<T> entriesLeftBehind = sqlDao.getEntriesLeftBehind(config.getMaxReDispatchCount(), now, reapingDate, config.getTableName());
+                final DateTime now = clock.getUTCNow();
+                final List<T> entriesLeftBehind = sqlDao.getEntriesLeftBehind(config.getMaxReDispatchCount(), now.toDate(), reapingDate, config.getTableName());
 
                 if (entriesLeftBehind.size() > 0) {
                     final Iterable<T> entriesToMove = Iterables.transform(entriesLeftBehind, new Function<T, T>() {
                         @Nullable
                         @Override
                         public T apply(@Nullable final T entry) {
-                            set(entry, "processingState", PersistentQueueEntryLifecycleState.REAPED);
+                            entry.setProcessingState(PersistentQueueEntryLifecycleState.REAPED);
                             return entry;
                         }
                     });
 
                     moveEntriesToHistoryFromTransaction(transactional, entriesToMove);
 
                     for (T entry : entriesLeftBehind) {
-                        set(entry, "creatingOwner", CreatorName.get());
-                        set(entry, "createdDate", clock.getUTCNow());
-                        set(entry, "processingState", PersistentQueueEntryLifecycleState.AVAILABLE);
+                        entry.setCreatedDate(now);
+                        entry.setProcessingState(PersistentQueueEntryLifecycleState.AVAILABLE);
+                        entry.setCreatingOwner(CreatorName.get());
                         insertEntryFromTransaction(transactional, entry);
                     }
 
-                    log.warn(String.format( "%s %s entries were reaped by %s",DB_QUEUE_LOG_ID ,entriesLeftBehind.size(), CreatorName.get()));
+                    log.warn(String.format( "{} {} entries were reaped by {}",DB_QUEUE_LOG_ID ,entriesLeftBehind.size(), CreatorName.get()));
```

Comment:
@wwjbatista Here and below: you also need to remove the `String.format` call:

```
log.warn("{} {} entries were reaped by {}",DB_QUEUE_LOG_ID ,entriesLeftBehind.size(), CreatorName.get());
 ```

See https://www.slf4j.org/faq.html#logging_performance.